core/app/components/PubCard.tsx

```diff
@@ -0,0 +1,146 @@
+		</>
+	);
+	const inner = link ? <Link href={link}>{iconWithTitle}</Link> : iconWithTitle;
+	return (
+		<Button variant="ghost" className="w-6 [&_svg]:size-6" asChild={!!link}>
+			{inner}
+		</Button>
+	);
+};
+
+export const PubCard = async ({
+	pub,
+	communitySlug,
+	stages,
+	actionInstances,
+}: {
+	pub: ProcessedPub<{ withPubType: true; withRelatedPubs: false; withStage: true }>;
+	communitySlug: string;
+	stages: CommunityStage[];
+	actionInstances: ActionInstances[];
+}) => {
+	return (
+		<Card
+			className="group flex items-center justify-between gap-2 rounded-md border border-gray-200 bg-white px-3 py-2"
+			data-testid={`pub-card-${pub.id}`}
+		>
+			<div className="flex min-w-0 flex-col space-y-[6px]">
+				<div className="flex flex-row gap-2 p-0 font-semibold leading-4">
+					{/* TODO: make filter by pub type */}
+					<Button variant="outline" className="h-[22px] rounded px-2 text-xs">
+						{pub.pubType.name}
+					</Button>
+					{pub.stage ? (
+						<Move
+							pubId={pub.id}
+							stageId={pub.stage.id}
+							communityStages={stages}
+							button={
+								<Button
+									variant="outline"
+									className="h-[22px] rounded-[104px] px-2 text-xs"
+								>
+									<FlagTriangleRightIcon strokeWidth="1px" />
+									{pub.stage.name}
+									<ChevronDown strokeWidth="1px" />
+								</Button>
+							}
+						/>
+					) : null}
+					<Button variant="outline" className="h-[22px] px-2 text-xs">
+						Relations
+						<ChevronDown strokeWidth="1px" />
+					</Button>
+				</div>
+				<CardTitle className="text-[15px] font-bold">
+					<h3 className="min-w-0 truncate">
+						<Link
+							href={`/c/${communitySlug}/pubs/${pub.id}`}
+							className="hover:underline"
+						>
+							{getPubTitle(pub)}
+						</Link>
+					</h3>
+				</CardTitle>
+				<CardDescription className="min-w-0 truncate">
+					<PubDescription pub={pub} />
+				</CardDescription>
+				<CardFooter className="flex gap-2 p-0 text-xs text-gray-600">
+					<div className="flex gap-1" title="Created at">
+						<Calendar size="16px" strokeWidth="1px" />
+						<span>{formatDateAsPossiblyDistance(new Date(pub.createdAt))}</span>
+					</div>
+					<div className="flex gap-1" title="Updated at">
+						<History size="16px" strokeWidth="1px" />
+						<span>{formatDateAsPossiblyDistance(new Date(pub.updatedAt))}</span>
+					</div>
+				</CardFooter>
+			</div>
+			<div className="opacity-0 transition-opacity duration-200 group-hover:opacity-100">
+				<div className="flex items-center gap-3 text-neutral-500">
+					<RemovePubButton
+						pubId={pub.id}
+						iconOnly
+						buttonText="Archive"
+						variant="ghost"
+						className="w-8 px-4 py-2 [&_svg]:size-6"
+						icon={<Trash strokeWidth="1px" />}
+					/>
+					{pub.stage ? (
+						<PubsRunActionDropDownMenu
+							actionInstances={actionInstances}
+							stage={pub.stage}
+							pubId={pub.id}
+							// @ts-ignore TODO: do we need this?
+							pageContext={{}}
```

Comment:
I couldn't quite figure out what the page context was used for. it would involve a lot of prop passing around to get it here, so I left it empty for now, but let me know if it should actually be filled out?